File: random_problems/1011.py
import math
N=int(input())
for i in range(N):
    x,y=map(int,input().split())
    d=y-x
    if d<4:
        print(d)
    else:
        sq=int(math.sqrt(d))
        if pow(sq,2)==d:
            print(sq*2-1)
        else:
            sq+=1
            if pow(sq,2)-sq>=d:
                print(sq*2-2)
            else:
                print(sq*2-1)

File: C_language/1011.c
#include<stdio.h>
#include<math.h>
int main()
{
	int N, i, x, y, d, sq;
	scanf("%d", &N);
	for (i = 0; i < N; i++) {
		scanf("%d %d", &x, &y);
		d = y - x;
		if (d < 4) printf("%d\n", d);
		else {
			sq = (int)sqrt(d);
			if (pow(sq, 2) == d) printf("%d\n", sq * 2 - 1);
			else {
				sq++;
				if (pow(sq, 2) - sq >= d) printf("%d\n", sq * 2 - 2);
				else printf("%d\n", sq * 2 - 1);
			}
		}
	}
}